sdk/identity/azure-identity/src/main/java/com/azure/identity/implementation/MsalToken.java

```diff
@@ -22,9 +22,10 @@
      *
      * @param msalResult the raw authentication result returned by MSAL
      */
-    public MsalToken(IAuthenticationResult msalResult) {
-        super(msalResult.accessToken(), OffsetDateTime.ofInstant(msalResult.expiresOnDate().toInstant(),
-            ZoneOffset.UTC));
+    public MsalToken(IAuthenticationResult msalResult, IdentityClientOptions options) {
+        super(msalResult.accessToken(), OffsetDateTime.ofInstant(
+                msalResult.expiresOnDate().toInstant().minus(options.getRefreshBeforeExpiry()), ZoneOffset.UTC)
+            .plusMinutes(2));
```

Comment:
Reverts the minus 2 minutes in AccessToken: https://github.com/Azure/azure-sdk-for-java/blob/master/sdk/core/azure-core/src/main/java/com/azure/core/credential/AccessToken.java#L22

Reply:
I feel like we are spreading the two minute concept out throughout the code base. Can you have it as a private static final value that is referred to everywhere?